### PR TITLE
Move to esbuild

### PR DESCRIPTION
**Pull request by kevinwhereby**
### Description

We are having some funny module resolution problems with rollup, so this pr aims to replace it with esbuild

Adds build steps for individual react/embed builds as well

### Gotchas
Esbuild can't emit typescript declarations, so we have to do it with `tsc`
Unfortunately, [tsc can't generate es6 concatenated declarations](https://github.com/microsoft/TypeScript/issues/2568) so we include the full directory structure... is this a problem? 

### To Test
- `yarn` to install the stuff
- `yarn build` to generate the builds and types
-  import it into a demo project

### Screenshots
Here's me importing just the react bundle like `import { useRoomConnection } from "@whereby.com/browser-sdk/react"`
image

and using the embed bundle in a script tag like
`<script type="module" src="./node_modules/@whereby.com/browser-sdk/dist/embed/index.esm.js"></script>`
image

### CODE REVIEW

File: tsconfig.build.json
Comment (havardholvik) [inline tsconfig.build.json:1]: Woud there be a way for us to use (or use most of) tsconfig.json in this config?

Comment (kevinwhereby) [reply tsconfig.build.json:1]: ahhh yes apparently tsconfig supports an `extends` key, let me add that

**Comment by kevinwhereby** [comment]
Hmm it seems the fullly bundled embed build doesn't work...
image

**Comment by kevinwhereby** [comment]
@havardholvik got the full embed bundle working, I think we're good to go now

**Comment by havardholvik** [comment]
> @havardholvik got the full embed bundle working, I think we're good to go now

Nice! Im doing some review and testing now, will be done shortly

File: src/lib/react/useRoomConnection.ts [hunk 1]
```diff
@@ -374,10 +374,7 @@ export type RoomConnectionRef = {
     _ref: RoomConnection;
 };
 
-export default function useRoomConnection(
-    roomUrl: string,
-    roomConnectionOptions: UseRoomConnectionOptions
-): RoomConnectionRef {
+export function useRoomConnection(roomUrl: string, roomConnectionOptions: UseRoomConnectionOptions): RoomConnectionRef {
```
Comment (kevinwhereby) [inline src/lib/react/useRoomConnection.ts:377]: This was causing webpack build errors when importing into a Create React App app.

**Pull request merged**